src/Containers/RecruitmentView/index.tsx

```diff
@@ -0,0 +1,192 @@
+
+  const allRecruitCostIron =
+    inputArcher * gameState.units.archer.trainingCost.iron +
+    inputSwordsman * gameState.units.swordsman.trainingCost.iron +
+    inputSpearman * gameState.units.spearman.trainingCost.iron +
+    inputAxeman * gameState.units.axeman.trainingCost.iron;
+
+  const allRecruitTime =
+    inputArcher * gameState.units.archer.trainingDuration +
+    inputSwordsman * gameState.units.swordsman.trainingDuration +
+    inputSpearman * gameState.units.spearman.trainingDuration +
+    inputAxeman * gameState.units.axeman.trainingDuration;
+
+  const unitData: RecruitmentContainerProps[] = [
+    {
+      name: 'archer',
+      unit: gameState.units.archer,
+      input: inputArcher,
+      setInput: setInputArcher,
+    },
+    {
+      name: 'swordsman',
+      unit: gameState.units.swordsman,
+      input: inputSwordsman,
+      setInput: setInputSwordsman,
+    },
+    {
+      name: 'axeman',
+      unit: gameState.units.axeman,
+      input: inputAxeman,
+      setInput: setInputAxeman,
+    },
+    {
+      name: 'spearman',
+      unit: gameState.units.spearman,
+      input: inputSpearman,
+      setInput: setInputSpearman,
+    },
+  ];
+
+  const units = unitData.map(({ name, unit, input, setInput }, key) => (
+    <RecruitmentContainer key={key} name={name} unit={unit} input={input} setInput={setInput} />
+  ));
+
+  const recruitUnits = async () => {
+    onLoading(true);
+    try {
+      const response = await fetch(`http://127.0.0.1:8000/api/v1/game/train_units/`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: localStorage.getItem('token') as string,
+        },
+        body: JSON.stringify({
+          units: [
+            { name: 'archer', count: inputArcher },
+            { name: 'axeman', count: inputAxeman },
+            { name: 'spearman', count: inputSpearman },
+            { name: 'swordsman', count: inputSwordsman },
+          ],
+        }),
+      });
+      if (response.ok) {
+        pushNotification('success', `Starting recruiting`);
+      } else {
+        const { errors } = await response.json();
+        console.log(errors);
+        Object.entries(errors).forEach(([key, value]) => {
+          pushNotification('warning', `${key}: ${(value as string[]).join(' and ')}`);
+        });
+      }
+    } catch (error) {
+      pushNotification('error', 'Server down', 'Please check your connection');
+    } finally {
+      onLoading(false);
+    }
+  };
+
+  return (
+    <TownHallWindow
+      open={open}
+      closable={true}
+      onCancel={() => setOpen(false)}
+      width={1000}
+      footer={false}
+      keyboard={true}
+      centered={true}
+    >
+      <Row align={'middle'} gutter={[0, 0]} style={{ margin: 'auto' }}>
+        <Col span={22}>
+          <Row align={'middle'} justify={'center'} gutter={[20, 0]}>
+            <Col span={4}>
+              <ResourcesNameTag>Unit</ResourcesNameTag>
```

Comment:
Can't we create some row component? In a different places I see rows with `margin: 'auto'`